src/core/net/nd6.hpp

```diff
@@ -832,33 +898,6 @@ class RouterAdvert
          */
         Error AppendFlagsExtensionOption(bool aStubRouterFlag);
 
-        /**
-         * Appends bytes from a given buffer to the RA message.
-         *
-         * @param[in] aBytes     A pointer to the buffer containing the bytes to append.
-         * @param[in] aLength    The buffer length.
-         *
-         * @retval kErrorNone    Bytes are appended successfully.
-         * @retval kErrorNoBufs  Insufficient available buffers to grow the message.
-         *
-         */
-        Error AppendBytes(const uint8_t *aBytes, uint16_t aLength);
-
-        /**
-         * Indicates whether or not the received RA message contains any options.
-         *
-         * @retval TRUE   If the RA message contains at least one option.
-         * @retval FALSE  If the RA message contains no options.
-         *
-         */
-        bool ContainsAnyOptions(void) const { return (mArray.GetLength() > sizeof(Header)); }
```

Comment:
Suggest keeping this method?
Any reason to remove?

Reply:
Hi, Abtink, the previous function `ContainsAnyOptions` will use the size of RA header which is defined above. And this function is only used for checking if a RA message has some options(if the size of message is larger than the header size). But now, I abstruct some function into the new class `Nd6TxMessage`, and for a `Nd6TxMessage`, the header will be appended, so I can not use `sizeof(Header)` to get the header size...

Reply:
How about also define a template function just like: 
```
template <typename ObjectType> bool ContainsAnyOptionsAfterHeader(const ObjectType &aHeader)
{
    return (mArray.GetLength() > sizeof(aHeader));
}
```

Reply:
I think we can still keep the `ContainsAnyOptions()` method within the subclass. 

You are right that we cannot determine the `Header` type in the base class `Nd6TxMessage`. But, it should be fine here, as we are defining it within the `TxMessage` nested type within the `RouterAdvert` class (so we know it an RA message). We have a nested `Header` defined within the same `RouterAdvert` class.

Reply:
Ok, I will make the function `ContainsAnyOptions` as a subclass function. And only implentment it for RouterAdvert Txmessage class.